auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -529,14 +527,94 @@ public Intent build() {
         }
 
         @VisibleForTesting()
+        public abstract FlowParameters getFlowParams();
+    }
+
+    /**
+     * Builder for the intent to start the reauthentication flow.
+     */
+    public final class ReauthIntentBuilder extends AuthIntentBuilder<ReauthIntentBuilder> {
+        private String mReauthReason;
+        private String mReauthEmail;
+
+        private ReauthIntentBuilder() {
+            super();
+        }
+
+        /**
+         * Set an explanation for why reauth was requested e.g. "To delete your account you must
+         * reauthenticate."
+         *
+         * @param reauthReason A string explaining why reauthentication was requested.
+         * @return
+         */
+        public ReauthIntentBuilder setReauthReason(String reauthReason) {
+            mReauthReason = reauthReason;
+            return this;
+        }
+
+        /**
+         * Sets the email address for which reauthentication will be tried.  If email accounts are
+         * enabled this MUST be set or email reauthentication will fail.
+         *
+         * @param reauthEmail The email address of the account for which reauthentication is being
+         * requested
+         */
+        public ReauthIntentBuilder setReauthEmail(String reauthEmail) {
```

Comment:
Is it possible to do re-auth and not use the currently signed in user?  In which case this email would be `FirebaseAuth.getInstance().getCurrentUser().getEmail()`  I feel like we could remove this setter and therefore reduce the chance of error/misunderstanding.